lib/parser/KOREScanner.l

```diff
@@ -97,51 +97,48 @@ yyin = in;
   if(YYSTATE != INITIAL)
     error(*loc, "Either a comment or string haven't been closed\n");
 
-  return TOKEN_EOF;
+  return token::TOKEN_EOF;
 }
 
 . { error(*loc, std::string("Unknown token \"") + yytext + std::string("\"\n")); }
 
 %%
 
-KOREScanner::KOREScanner(const char *filename) {
-  if (!(in = fopen(filename, "r"))) {
+KOREScanner::KOREScanner(std::string filename) {
+  if (!(in = fopen(filename.c_str(), "r"))) {
     std::cerr << "Cannot read file: " << filename << "\n";
     exit(1);
   }
+  yylex_init(&scanner);
```

Comment:
I am a little confused as to how the parser state is getting initialized here. This `scanner` variable is just a `void *` right?

Reply:
yylex_init allocates some storage and then writes the address of that storage to this pointer.

Reply:
OK I see. I think I get it now.